test/e2e/workflow/deployment_test.go

```diff
@@ -438,6 +438,57 @@ var _ = Describe("NetworkAddonsConfig", func() {
 					}, 5*time.Minute, time.Second).Should(BeNil(), "Timed out waiting for nmstate-operator daemonset")
 				})
 
+				Context("when NAC is updated", func() {
+					BeforeEach(func() {
```

Comment:
correct me if I'm wrong, but I don't think you are actually updating the `configSpec` . You're missing this:
```
UpdateConfig(gvk, configSpec)
CheckConfigCondition(gvk, ConditionAvailable, ConditionTrue, 15*time.Minute, CheckDoNotRepeat)
```

Reply:
I would also add `By("By adding a unique SelfSignConfiguration configuration")` or something the like

Reply:
> correct me if I'm wrong, but I don't think you are actually updating the configSpec . You're missing this:

The config is created at `JustBeforeEach` called in 
https://github.com/kubevirt/cluster-network-addons-operator/pull/1371/commits/08d160a48ed4b089cc654761fd1112b3bb3a3e68#diff-cf2d447e13fe89b62ab6c7d5e1e9b8b4e931b2bd88a15e5777f829650b92b66bR421

This is the initial setting with which networkAddonsConfig is created

Reply:
ah, learned something new, cool! :)